stake-pool/cli/src/main.rs

```diff
@@ -374,66 +344,219 @@ fn command_create_pool(
             validator_list_balance,
             validator_list_size as u64,
             &spl_stake_pool::id(),
         ),
-        // Account for the stake pool
-        system_instruction::create_account(
-            &config.fee_payer.pubkey(),
-            &stake_pool_keypair.pubkey(),
-            stake_pool_account_lamports,
-            get_packed_len::<StakePool>() as u64,
-            &spl_stake_pool::id(),
-        ),
-        // Initialize stake pool
+    ];
+
+    let signers = vec![config.fee_payer.as_ref(), validator_list_keypair];
+    let transaction = checked_transaction_with_signers(config, &instructions, &signers)?;
+
+    println!("Creating validator list account {}", validator_list_keypair.pubkey());
+    send_transaction(config, transaction)?;
+    Ok(())
+}
+
+fn initialize_stake_pool(
+    config: &Config,
+    stake_pool_keypair: &Keypair,
+    validator_list_keypair: &Keypair,
+    reserve_keypair: &Keypair,
+    mint_keypair: &Keypair,
+    pool_fee_account: &Pubkey,
+    deposit_authority: Option<Keypair>,
+    epoch_fee: Fee,
+    withdrawal_fee: Fee,
+    deposit_fee: Fee,
+    referral_fee: u8,
+    max_validators: u32,
+    withdraw_authority: &Pubkey,
+) -> CommandResult {
+    let stake_pool_account_info = config.rpc_client.get_account(&stake_pool_keypair.pubkey());
+
+    let stake_pool_account_lamports = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(get_packed_len::<StakePool>())?;
+        
+        
+    if let Ok(account) = stake_pool_account_info {
+        let stake_pool_manager = get_stake_pool(&config.rpc_client, &stake_pool_keypair.pubkey()).unwrap().manager;
+        if account.owner == spl_stake_pool::id() {
+            if stake_pool_manager == config.manager.pubkey() {
+                println!("Stake pool account {} already exists and is initialized", stake_pool_keypair.pubkey());
+                return Ok(());
+            } else {
+                println!("Stake pool account {} already exists but is not initialized", stake_pool_keypair.pubkey());
+            }
+        }
+    } else {
+        // Create the stake pool account if it doesn't exist
+        let create_instructions = vec![
+            system_instruction::create_account(
+                &config.fee_payer.pubkey(),
+                &stake_pool_keypair.pubkey(),
+                stake_pool_account_lamports,
+                get_packed_len::<StakePool>() as u64,
+                &spl_stake_pool::id(),
+            ),
+        ];
+
+        let mut create_signers = vec![
+            config.fee_payer.as_ref(),
+            stake_pool_keypair,
+        ];
+
+        unique_signers!(create_signers);
+        let create_transaction = checked_transaction_with_signers(config, &create_instructions, &create_signers)?;
+
+        println!(
+            "Creating stake pool account {}",
+            stake_pool_keypair.pubkey(),
+        );
+        send_transaction(config, create_transaction)?;
```

Comment:
Let's put the creation and initialization of the stake pool in the same transaction (along with the creation of the validator list, as mentioned in another comment) -- more transactions just means more opportunities for things to go wrong